src/acl.c

```diff
@@ -2188,9 +2187,24 @@ void aclCommand(client *c) {
             struct redisCommand *cmd = dictGetVal(de);
             if (cmd->flags & CMD_MODULE) continue;
             if (cmd->acl_categories & cflag) {
-                addReplyBulkCString(c,cmd->name);
+                addReplyBulkCBuffer(c, cmd->fullname, sdslen(cmd->fullname));
                 arraylen++;
             }
+
+            /* Handle subcommands */
+            if (cmd->subcommands_dict) {
```

Comment:
generally speaking, i think i prefer if all places that iterate commands would use a recursive function instead of copying the logic... @oranagra WDYT?

Reply:
i think that if i'ts just a matter of calling one or two functions or some simple `if`, then there's no problem copying it.
if there's a lot of logic, then we should avoid repeating it, and instead add a function to do that, or use a recursion.
bottom line, each case for itself IMHO.